### PR TITLE
Create HoC Live Spanish banner

### PR DESCRIPTION
**Pull request by hacodeorg**
[Task FND-1315](https://codedotorg.atlassian.net/browse/FND-1315): Create HoC Live Spanish banner on the HoC landing page ([spec](https://docs.google.com/document/d/10INUk_aUhLsHRdoQDKxGpMr86GYZSV5sWNMJINN5Avk/edit)).

## Screenshots
On desktop, the banner will look like this:

Screen Shot 2020-10-06 at 6 15 08 PM

On a smaller screen such as the iPhone 6 (375x667):

Screen Shot 2020-10-07 at 10 45 07 AM

The banner can also be resized:

Resize HoC banner - fixed

## Testing story
Verify that the banner shows up for Spanish-speaking users
- In Spain: http://localhost.hourofcode.com:3000/es
- In Argentina (LATAM): http://localhost.hourofcode.com:3000/ar
- In US but use Spanish: http://localhost.hourofcode.com:3000/us/es
- In US but use LATAM Spanish: http://localhost.hourofcode.com:3000/us/la

Verify that the banner doesn't show up for non-Spanish speaking users
- In US (default = English): http://localhost.hourofcode.com:3000/us/
- In Brazil (LATAM, default = Portugese) http://localhost.hourofcode.com:3000/br/
- In Argentina (LATAM) but use English: http://localhost.hourofcode.com:3000/ar/en

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by hacodeorg** [comment]
This might be a basic styling technique, I don't know how to keep the height of the left and right box (image and message) the same when resizing. Any suggestions?

Resize HoC banner

**Comment by dju90** [comment]
Can we pre-emptively add the Spanish translation to es.yml and la.yml? I can be responsible for proactively uploading the changes to en.yml to Crowdin.

File: pegasus/sites.v3/hourofcode.com/public/index.haml [hunk 1]
```diff
@@ -101,6 +101,23 @@ social:
 %br/
 
 .container
+  -# HoC Live Spanish banner
+  - if @language == 'es' || @language == 'la'
+    #hoclivebanner.row
+      %div.banner-box
+        %div.banner-image-box
+          %img.banner-image{src: '/images/hora-del-codigo-en-vivo.png', alt: hoc_s(:hoc_live)}
```
Comment (Hamms) [inline pegasus/sites.v3/hourofcode.com/public/index.haml:109]: does this image need to be added?

File: pegasus/sites.v3/hourofcode.com/public/css/front-page.css [hunk 2]
```diff
@@ -96,6 +96,52 @@ a.category-link {
   margin-right: 10px;
 }
 
+.banner-box {
+  width: 100%;
+  height: fit-content;
```
Comment (Hamms) [inline pegasus/sites.v3/hourofcode.com/public/css/front-page.css]: note that fit-content is not supported by IE, but is also I think not necessary here

**Review by Hamms — COMMENTED** [review]
The best way that I'm aware of to get sibling elements to be height-locked is to use `display: table-row;` on the parent element and `display: table-cell;` on the children. You'll also want to get rid of the current `float` directives:

```html
<div class="banner-box" style="display: table-row;">
  <div class="banner-image-box" style="display: table-cell; float: none;">
    ...
  </div>
  <div class="banner-message-box" style="display: table-cell; float: none;">
    ...
  </div>
</div>
```

File: pegasus/sites.v3/hourofcode.com/public/index.haml [hunk 3]
```diff
@@ -101,6 +101,23 @@ social:
...
+        %div.banner-message-box
+          %div.banner-title
+            = hoc_s(:hoc_live_sign_up_title)
+          %div.banner-message
+            = hoc_s(:hoc_live_sign_up_message)
+          %a#signuplivebutton.ctabuttonatag{href: 'http://go.pardot.com/l/153401/2020-09-30/nwvb6j', target: '_blank'}
```
Comment (Hamms) [inline pegasus/sites.v3/hourofcode.com/public/index.haml]: Important note: `target: _blank` is a security vulnerability. If we absolutely _must_ open links in new tabs, there are some things we can do to mitigate those security risks, but in this case I don't see a need for us to do that. Situations in which opening a link in a new tab are necessary are mostly those situations in which there are interactions on the page that could be lost if the user unexpectedly navigates away; things like a page with a long form to be filled out or, say, a page with a code editor that may lose progress.

The hourofcode.com homepage is a static page, so we should just leave this anchor as normal.

For more information, see:

- https://medium.com/sedeo/how-to-fix-target-blank-a-security-and-performance-issue-in-web-pages-2118eba1ce2f
- https://www.jitbit.com/alexblog/256-targetblank---the-most-underestimated-vulnerability-ever/
- https://blog.dareboost.com/en/2017/03/target-blank-links-rel-noopener-performance-security/

**Review by Hamms — CHANGES_REQUESTED** [review]

**Comment by dju90** [comment]
> This might be a basic styling technique, I don't know how to keep the height of the left and right box (image and message) the same when resizing. Any suggestions?

fwiw, code studio homepage banners suffer from the same mismatched height issue _when the text on the right is too long_.
image

The technique they use there for smaller screen sizes is to just get rid of the image on the left when width < 993px:
image

File: pegasus/sites.v3/hourofcode.com/public/index.haml (hunk 3 above)
Comment (dju90) [reply pegasus/sites.v3/hourofcode.com/public/index.haml]: I agree that this doesn't have to open on a new tab :)

**Comment by dju90** [comment]
There is already a string key for "Join us" in en.yml - `front_join_us_button`. Should we use that instead of adding a new string key (`hoc_live_join_us`)? Not sure what the best practice is here :)

File: pegasus/sites.v3/hourofcode.com/public/css/front-page.css (hunk 2 above)
Comment (hacodeorg) [reply pegasus/sites.v3/hourofcode.com/public/css/front-page.css]: Removed.

File: pegasus/sites.v3/hourofcode.com/public/index.haml (hunk 1 above)
Comment (hacodeorg) [reply pegasus/sites.v3/hourofcode.com/public/index.haml:109]: Yes, I'll add it.

File: pegasus/sites.v3/hourofcode.com/public/index.haml (hunk 3 above)
Comment (hacodeorg) [reply pegasus/sites.v3/hourofcode.com/public/index.haml]: Removed. Thank you for the detailed explanation! It's really helpful.

**Comment by hacodeorg** [comment]
> Can we pre-emptively add the Spanish translation to es.yml and la.yml? I can be responsible for proactively uploading the changes to en.yml to Crowdin.

^ Added!

> fwiw, code studio homepage banners suffer from the same mismatched height issue _when the text on the right is too long_.

^ Elijah's recommendation using `display: table` and `display: table-cell` fixes the problem. Furthermore, the elements inside each cell have to be aligned using `vertical-align` and `text-align`.

> There is already a string key for "Join us" in en.yml - `front_join_us_button`. Should we use that instead of adding a new string key (`hoc_live_join_us`)? Not sure what the best practice is here :)

^ The 2 "Join Us" buttons are for two different actions. The original "Join Us" button brings users to the registration form at the middle of the page. The new "Join Us" button brings user to a Pardot landing page. My instinct is we should separate them because they can change in different ways in the future.

**Review by Hamms — APPROVED** [review]

**Pull request merged**